lib/components/overlays/quick_settings/quick_settings_overlay.dart

```diff
@@ -276,19 +277,22 @@ class QsMain extends StatelessWidget {
                       mainAxisAlignment: MainAxisAlignment.spaceBetween,
                       children: [
                         QsToggleButton(
-                          title: LSX
+                          title: strings
                               .quicksettingsOverlay.quickControlsLanguageTitle,
                           //TODO Fix this
                           subtitle: context.locale.toLanguageTag(),
                           icon: Icons.language_rounded,
                           value: true,
                           onPressed: () {
-                            final int index =
-                                Locales.supported.indexOf(context.locale);
-                            if (index + 1 < Locales.supported.length) {
-                              context.setLocale(Locales.supported[index + 1]);
+                            final int index = locales.supportedLocales
+                                .indexOf(context.locale.toIntlLocale());
+                            if (index + 1 < locales.supportedLocales.length) {
+                              context.locale = locales
+                                  .supportedLocales[index + 1]
+                                  .toFlutterLocale();
```

Comment:
As above, use context.supportedLocales to avoid calling .toFlutterLocale() and .toIntlLocale()